hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/OMClientRequestUtils.java

```diff
@@ -47,4 +60,42 @@ public static void checkClientRequestPrecondition(
           OMException.ResultCodes.INTERNAL_ERROR);
     }
   }
+
+  public static boolean isSnapshotBucket(OMMetadataManager omMetadataManager,
+      OmKeyInfo keyInfo) throws IOException {
+
+    String dbSnapshotBucketKey = omMetadataManager.getBucketKey(
+        keyInfo.getVolumeName(), keyInfo.getBucketName())
+        + OM_KEY_PREFIX;
+
+    return checkInSnapshotCache(omMetadataManager, dbSnapshotBucketKey) ||
+        checkInSnapshotDB(omMetadataManager, dbSnapshotBucketKey);
+  }
+
+  private static boolean checkInSnapshotDB(OMMetadataManager omMetadataManager,
+      String dbSnapshotBucketKey) throws IOException {
+    try (TableIterator<String, ? extends Table.KeyValue<String, SnapshotInfo>>
+        iterator = omMetadataManager.getSnapshotInfoTable().iterator()) {
+      iterator.seek(dbSnapshotBucketKey);
+      return iterator.hasNext() && iterator.next().getKey()
+          .startsWith(dbSnapshotBucketKey);
+    }
+  }
+
+  private static boolean checkInSnapshotCache(
+      OMMetadataManager omMetadataManager,
+      String dbSnapshotBucketKey) {
+    Iterator<Map.Entry<CacheKey<String>, CacheValue<SnapshotInfo>>> cacheIter =
+        omMetadataManager.getSnapshotInfoTable().cacheIterator();
+
+    while (cacheIter.hasNext()) {
+      Map.Entry<CacheKey<String>, CacheValue<SnapshotInfo>> cacheKeyValue =
+          cacheIter.next();
+      String key = cacheKeyValue.getKey().getCacheKey();
+      if (key.startsWith(dbSnapshotBucketKey)) {
+        return true;
+      }
```

Comment:
Not only do we need to check for cache entry existence. We also need to check whether the entry is a tombstone entry (cache entry value is `null`) in cache.

If it is a tombstone, should return `false`.